File: GoL/Model/CellState.m
//
//  CellState.m
//  GoL
//
//  Created by Georg Meyer on 27.04.14.
//
//

#import "CellState.h"

@interface CellState ()

@property (nonatomic) BOOL currentState;

@property (nonatomic) BOOL nextState;

@end


@implementation CellState

- (BOOL)isAlive {
    return self.currentState;
}

- (void)applyNextState {
    self.currentState = self.nextState;
}

- (void)storeNextState:(BOOL)value {
    self.nextState = value;
}

+ (CellState *)createAlive {
    return [[self alloc] initWithCurrentState:YES];
}

+ (CellState *)createDead {
    return [[self alloc] initWithCurrentState:NO];
}

- (CellState *)initWithCurrentState:(BOOL)currentState {
    self = [super init];
    if (self) {
        self.currentState = currentState;
    }
    return self;
}

@end

File: GoL/Model/CellState.swift
//
//  CellState.swift
//  GoL
//
//  Created by Georg Meyer on 04.06.14.
//
//

import Foundation

class CellState {
    var currentState = false
    var nextState = false
    
    init(currentState : Bool){
        self.currentState = currentState
    }
    
    func isAlive() -> Bool {
        return currentState
    }
    
    func applyNextState(){
        currentState = nextState
    }
    
    func storeNextState(nextState : Bool){
        self.nextState = nextState
    }
    
    func die(){
        currentState = false
    }
    
    func revive(){
        currentState = true
    }
    
    class func createAlive() -> CellState{
        return CellState(currentState: true)
    }
    
    class func createDead() -> CellState{
        return CellState(currentState: false)
    }
}